tests/queries/quote_rewards_test_db.sql

```diff
@@ -0,0 +1,67 @@
+DROP TABLE IF EXISTS orders;
+DROP TYPE IF EXISTS OrderKind;
+DROP TABLE IF EXISTS order_quotes;
+DROP TABLE IF EXISTS trades;
+
+
+-- orders table
+CREATE TYPE OrderKind AS ENUM ('buy', 'sell');
+
+CREATE TABLE IF NOT EXISTS orders
+(
+    uid bytea PRIMARY KEY,
+    sell_amount numeric(78,0) NOT NULL,
+    buy_amount numeric(78,0) NOT NULL,
+    fee_amount numeric(78,0) NOT NULL,
+    kind OrderKind NOT NULL,
+    partially_fillable boolean NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS order_quotes
+(
+  order_uid bytea PRIMARY KEY,
+  sell_amount numeric(78, 0) NOT NULL,
+  buy_amount numeric(78, 0) NOT NULL,
+  solver bytea NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS trades
+(
+  block_number bigint NOT NULL,
+  log_index bigint NOT NULL,
+  order_uid bytea NOT NULL,
+  sell_amount numeric(78, 0) NOT NULL,
+  buy_amount numeric(78, 0) NOT NULL,
+
+  PRIMARY KEY (block_number, log_index)
+);
+
+
+TRUNCATE orders;
+TRUNCATE order_quotes;
+TRUNCATE trades;
+
+
+INSERT INTO orders (uid, sell_amount, buy_amount, fee_amount, kind, partially_fillable)
+VALUES ('\x01'::bytea, 95000000, 94000000000000000000, 5000000, 'sell', 'f'), -- normal sell market order
+('\x02'::bytea, 101000000, 100000000000000000000, 5000000, 'buy', 'f'), -- normal buy market order
+('\x03'::bytea, 100000000, 100000000000000000000, 0, 'sell', 't'), -- partially fillable sell limit order
+('\x04'::bytea, 100000000, 100000000000000000000, 0, 'buy', 't'), -- partially fillable buy limit order
+('\x05'::bytea, 100000000, 94000000000000000000, 0, 'sell', 'f'), -- in market sell limit order
+('\x06'::bytea, 106000000, 100000000000000000000, 0, 'buy', 'f'); -- in market buy limit order
+
+INSERT INTO order_quotes (order_uid, sell_amount, buy_amount, solver)
+VALUES ('\x01'::bytea, 95000000, 95000000000000000000, '\x01'::bytea),
+('\x02'::bytea, 101000000, 100000000000000000000, '\x02'::bytea),
+('\x03'::bytea, 100000000, 95000000000000000000, '\x03'::bytea),
+('\x04'::bytea, 105000000, 100000000000000000000, '\x03'::bytea),
+('\x05'::bytea, 100000000, 95000000000000000000, '\x03'::bytea),
```

Comment:
Let's look at order \x05. I think that we should have order_quotes.sell_amount < orders.sell_amount, as the quote takes into account the fee.

Reply:
I just checked our database and it seems that for limit orders, the sell amount is not reduced by some fee amount. E.g.
`select * from orders where uid = '\xD3C1CE97725452DFF58BDC755DDE48E0A7420046075E092BCBC2F397484A287DBDBCE09CF017D8BD51C88B988297B594BD2D3F1065A5E83A'` and `select * from order_quotes where order_uid = '\xD3C1CE97725452DFF58BDC755DDE48E0A7420046075E092BCBC2F397484A287DBDBCE09CF017D8BD51C88B988297B594BD2D3F1065A5E83A'` both return the same sell amount for some order.

Reply:
Yeah you are probably right, although I find this a bit confusing and want to look into how the `order_quotes` table is created. Somewhat irrelevant to this PR though